### PR TITLE
Autoinstall clarification

### PR DESCRIPTION
**Pull request by holmanb**
```
autoinstall: clarify docs for users
```
Add autoinstall pointer to subiquity docs to FAQ.

Remove cc_ubuntu_autoinstall from docs module definitions, since it doesn't give users actionable information.

Leave the module definition intact in the code, since it does add context that may be valuable for developers.

The net affect of this is PR that future searches of the website for "autoinstall" will lead users to the FAQ (and then to subiquity docs), which makes sense because we don't have autoinstall-specific documentation in the cloud-init docs.

### CODE REVIEW

**Review by TheRealFalcon — APPROVED** [review]
LGTM, but lets get a +1 from @blackboxsw too

File: doc/rtd/topics/faq.rst
Comment (blackboxsw) [inline doc/rtd/topics/faq.rst:208]: YAML should be caps in docs I think.

File: doc/rtd/topics/faq.rst
Comment (blackboxsw) [inline doc/rtd/topics/faq.rst:209]: Let's decorate these a bit as they are reserved keys:
```suggestion
however keys such as ``autoinstall``, ``preruncmd``, and ``postruncmd`` are not used by
```

File: doc/rtd/topics/faq.rst
Comment (blackboxsw) [inline doc/rtd/topics/faq.rst:206]: It it worth us linking to Juju's model config docs too? https://discourse.charmhub.io/t/model-config-cloudinit-userdata/512

File: doc/rtd/topics/faq.rst
Comment (blackboxsw) [inline doc/rtd/topics/faq.rst:212]: Shall we link to the support channels here? For Juju maybe https://discourse.charmhub.io/  and for autoinstaller either #subiquity channel on IRC or #ubuntu-server maybe. We should check on that with subiquity devs.
@dbungert any preferences on what mediums autoinstall questions get steered?

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @holmanb for making this a clearer line in documentation. Let's round out the references to other projects so folks go to the right place/channel to seek answers.
minor nits otherwise

File: doc/rtd/topics/faq.rst
Comment (s-makin) [reply doc/rtd/topics/faq.rst:206]: +1 to that

File: doc/rtd/topics/faq.rst
Comment (s-makin) [reply doc/rtd/topics/faq.rst:212]: +1 to linking the support channels

**Review by s-makin — APPROVED** [review]
Overall LGTM, I'm assuming that Chad will add his approval after the links are added, but I'll add mine now.

File: doc/rtd/topics/faq.rst
Comment (dbungert) [reply doc/rtd/topics/faq.rst:212]: There isn't a #subiquity channel today on IRC and based on the average feedback I have seen on IRC I don't plan to start one.  Others are certainly welcome.

Discourse and #ubuntu-server are the way to go today.

Comment (holmanb) [reply doc/rtd/topics/faq.rst]: +1 Thanks, I've added that to this PR.

File: doc/rtd/topics/faq.rst
Comment (holmanb) [inline doc/rtd/topics/faq.rst:154]: I removed this since it doesn't make sense where it is. I think this may have previously preceded the section titled "How can I test cloud-init locally before deploying to the cloud?", but I don't think it adds anything there, so I propose deleting it.

**Comment by holmanb** [comment]
Thanks @blackboxsw and @s-makin for the reviews. I updated the links, and deleted some lost text.

Ready for review.

**Review by blackboxsw — APPROVED** [review]
+1 thanks for this @holmanb

**Pull request merged**
